docs2/site/docs/getting-started/dependency-injection.md

```diff
@@ -42,6 +42,69 @@ public class StarWarsSchema : GraphQL.Types.Schema
+* `IDocumentCache` - an implemenation which does not cache documents
+
+A list of the available extension methods is below:
+
+| Method    | Description / Notes | Library |
+|-----------|---------------------|---------|
+| `AddClrTypeMappings`    | Scans the specified assembly for graph types intended to represent CLR types and registers them within the schema | |
+| `AddComplexityAnalyzer` | Enables the complexity analyzer and configures its options | |
+| `AddDataLoader`         | Registers classes necessary for data loader support | GraphQL.DataLoader |
+| `AddDocumentCache<>`    | Registers the specified document caching service | |
+| `AddDocumentExecuter<>` | Registers the specified document executer; useful when needed to change the execution strategy utilized | |
+| `AddDocumentListener<>` | Registers the specified document listener and configures execution to use it | |
+| `AddDocumentWriter<>`   | Registers the specified document writer | |
+| `AddErrorInfoProvider`  | Registers a custom error info provider or configures the default error info provider | |
+| `AddGraphTypes`         | Scans the specified assembly for graph types and registers them within the DI framework | |
+| `AddMemoryCache`        | Registers the memory document cache and configures its options | GraphQL.MemoryCache |
+| `AddMetrics`            | Registers and enables metrics depending on the supplied arguments | |
+| `AddMiddleware<>`       | Registers the specified middleware and configures it to be installed during schema initialization | |
+| `AddNewtonsoftJson`     | Registers the document writer that uses Newtonsoft.Json as its underlying JSON serialization engine | GraphQL.NewtonsoftJson |
+| `AddSchema<>`           | Registers the specified schema | |
+| `AddSelfActivatingSchema<>` | Registers the specified schema which will create instances of unregistered graph types during initialization | |
+| `AddSubscriptionDocumentExecuter` | Registers the document executer that has subscription support | GraphQL.SystemReactive | |
+| `AddSystemTextJson`     | Registers the document writer that uses System.Text.Json as its underlying JSON serialization engine | GraphQL.SystemTextJson |
+| `AddValidationRule<>`   | Registers the specified validation rule and configures it to be used at runtime | |
+| `ConfigureExecution`    | Configures execution options at runtime | |
+| `ConfigureSchema`       | Configures schema options when the schema is initialized | |
+| `Configure<TOptions>`   | Used by extension methods to configures an options class within the DI framework | |
+| `Register`              | Used by extension methods to register services within the DI framework | |
+| `TryRegister`           | Used by extension methods to register services within the DI framework when they have not already been registered | |
+
+The above methods will register the specified services typically as singletons unless otherwise specified. Graph types and middleware is registered
```

Comment:
Graph types and middleware is registered
->
Graph types and middlewares are registered

Reply:
"middleware" comes from the words "middle" and "software" and is both singular and plural, just like "software" or "sheep" or "deer".  It is improper to write "softwares" or "middlewares" or "sheeps" or "deers".

Reply:
🤔 OK. What about `is` -> `are` ?

Reply:
Might sound a little awkward but is probably correct

Reply:
```suggestion
The above methods will register the specified services typically as singletons unless otherwise specified. Graph types and middleware are registered
```

Reply:
#2792